File: paddler_types/src/pooling_type.rs
use serde::Deserialize;
use serde::Serialize;

#[derive(Clone, Debug, Deserialize, Serialize)]
#[repr(i8)]
pub enum PoolingType {
    Unspecified = -1,
    None = 0,
    Mean = 1,
    Cls = 2,
    Last = 3,
    Rank = 4,
}

File: paddler_client_javascript/src/schemas/PoolingType.ts
import { z } from "zod";

export const PoolingTypeSchema = z.enum([
  "Cls",
  "Last",
  "Mean",
  "None",
  "Rank",
  "Unspecified",
]);

export type PoolingType = z.infer<typeof PoolingTypeSchema>;
